src/simmate/database/base_data_types/structure.py

```diff
@@ -271,3 +284,85 @@ def to_toolkit(self) -> ToolkitStructure:
         Converts the database object to toolkit Structure object.
         """
         return ToolkitStructure.from_database_object(self)
+    
+    @classmethod
+    def filter_similarity(cls, structure, method="L2", cutoff=0.8):
```

Comment:
This is a good start. I have a more developed version of this with several improvements that I'll share once you flag the MR for review. If this is working for you now, just leave it -- if it's giving you issues, let me know and just hold off and let me refactor/update it

Reply:
Sounds good. I've only been able to test it for the sqlite3 backend and it works fine for that, but I'll the postgres side once I get it working.

Reply:
I removed you're `postgress_connect()` and will show why in a follow up MR. Effectively, [this is the preferred way](https://docs.djangoproject.com/en/5.1/topics/db/sql/) to do raw SQL queries in django, but there's an even better approach using annotations that I'll draft when I have time tomorrow